js/script1.js

```diff
@@ -0,0 +1,226 @@
+        document.getElementById('star3').classList.add("fa-star-o");
+        stars = 2;
+    }
+    else if (moves > 18) {
+        document.getElementById('star2').classList.remove("fa-star")
+        document.getElementById('star2').classList.add("fa-star-o");
+        stars = 1;
+    }
+
+}
+function endGame(moves) {
+    starRating(moves);
+    swal({
+        allowEscapeKey: false,
+        allowOutsideClick: false,
+        title: 'Congratulations! You have Won!',
+        text: 'With ' + moves + ' steps and ' + stars + ' Stars.\n Woooooo! \n Time Taken: ' + minutes + ' Minutes - ' + seconds + ' Seconds',
+        type: 'success',
+        confirmButtonColor: '#82d8e7',
+        confirmButtonText: 'Play again!'
+    }).then(function () {
+        clearInterval(timer_interval);
+        setInitial();
+    })
+
+
+
+
+
+}
+function clearIcon() {
+
+    //clear all className added in i tags to remove any icons setted before and add intial class
+
+    let aLL_I_Tags = document.querySelectorAll('.container')[0].querySelectorAll('i.fa');
+    aLL_I_Tags.forEach((element) => {
+        element.className = "fa";
+    })
+
+}
+function changeCardState(firstCard, secondCard) {
+    // add matched class
+    firstCard.classList.add('matched');
+    secondCard.classList.add('matched');
+    // add confirm class to  change color of flipped side.
+    firstCard.getElementsByClassName('back')[0].classList.add('confirm');
+    secondCard.getElementsByClassName('back')[0].classList.add('confirm');
+}
+function setInitial() {
+    game_started = false;
+    moves = 0;
+    starRating(moves);
+    clearIcon();
+    shuffleIcons(icons_class);
+    setIcons(icons_class);
+
+    seconds = 0;
+    minutes = 0;
+    document.getElementById('move_value').innerText = moves;
+    document.getElementById('seconds').innerText = seconds;
+    document.getElementById('minutes').innerText = minutes;
+    let cards = document.querySelectorAll('.card');
+    [...cards].forEach((card, index) => {
+        card.setAttribute("id", index);
+        card.className = ("card");
+
+        let backCard = card.getElementsByClassName('back')[0];
+        if (backCard.classList.contains("confirm"))
+            backCard.classList.remove("confirm");
+
+    });
+
+
+
+
+
+
+}
+function setCardClick() {
+    let cards = document.querySelectorAll('.card');
+    [...cards].forEach((card, index) => {
+        card.addEventListener('click', function (event) {
+
+            card.classList.add('is-flipped');
+            //check if game is started or not
+            if (game_started == false) {
+                setTimer();
+                openedCardIds.push(index);
+                game_started = true;
+                return;
+            }
+            //if clicked card is already card or not
+            if (card.classList.contains("matched")) {
+                return;
+            }
+            //if clciked card is first card.
+            if (openedCardIds.length == 0) {
```

Comment:
Shouldn't the openedCardIds.length =0 when the game has not started ? what was the problem if we combine if's of line 150 & 161. When are we coming inside this if. ?